libs/ledger-live-common/src/market/v2/useMarketDataProvider.ts

```diff
@@ -0,0 +1,128 @@
+  const { data: supportedCounterCurrencies } = useSupportedCounterCurrencies();
+  const { data: supportedCurrencies } = useSupportedCurrencies();
+
+  const liveCoinsList = useMemo(
+    () =>
+      supportedCurrencies?.filter(({ id }) => liveCompatibleIds.includes(id)).map(({ id }) => id),
+    [liveCompatibleIds, supportedCurrencies],
+  );
+
+  return {
+    supportedCounterCurrencies,
+    supportedCurrencies,
+    liveCoinsList,
+  };
+}
+
+export function useCurrencyChartData({ id, counterCurrency, ranges }: MarketCurrencyRequestParams) {
+  const results = useQueries({
+    queries: (ranges || []).map(range => ({
+      queryKey: [QUERY_KEY.CurrencyChartData, id, counterCurrency, range],
+      queryFn: () => fetchCurrencyChartData({ counterCurrency, range, id }),
+      refetchInterval: REFETCH_TIME,
+      staleTime: REFETCH_TIME,
+    })),
+  });
+
+  return results;
+}
+
+export function useCurrencyData({ id, counterCurrency, ranges }: MarketCurrencyRequestParams) {
+  const results = useQueries({
+    queries: (ranges || []).map(range => ({
+      queryKey: [QUERY_KEY.CurrencyData, id, counterCurrency, range],
+      queryFn: () => fetchCurrencyData({ counterCurrency, range, id }),
+      refetchInterval: REFETCH_TIME,
+      staleTime: REFETCH_TIME,
+    })),
+  });
+
+  const resultCurrency = useQuery({
+    queryKey: [QUERY_KEY.CurrencyDataRaw, id],
+    queryFn: () => fetchCurrency({ id }),
+    refetchInterval: REFETCH_TIME,
+    staleTime: REFETCH_TIME,
+  });
+
+  return { currencyDataByRanges: results, currencyInfo: resultCurrency };
+}
+
+export function useSupportedCounterCurrencies() {
+  const { isPending, error, data, isFetching } = useQuery({
+    queryKey: [QUERY_KEY.SupportedCounterCurrencies],
+    queryFn: () => supportedCounterCurrencies(),
+  });
+
+  return {
+    isPending,
+    error,
+    data,
+    isFetching,
+  };
+}
+
+export function useSupportedCurrencies() {
+  const { isPending, error, data, isFetching } = useQuery({
+    queryKey: [QUERY_KEY.SupportedCurrencies],
+    queryFn: () => getSupportedCoinsList(),
+  });
+
+  return {
+    isPending,
+    error,
+    data,
+    isFetching,
+  };
+}
+
+export function useMarketData(props: MarketListRequestParams): MarketListRequestResult {
+  const results = useQueries({
+    queries: Array.from({ length: props.page ?? 1 }, (_, i) => i + 1).map(page => ({
+      queryKey: [
+        QUERY_KEY.MarketData,
+        { ...props, page, liveCoinsList: [], supportedCoinsList: [] },
+      ],
+      queryFn: () => fetchList({ ...props, page }),
+      refetchInterval: props.refreshTime ?? REFETCH_TIME,
+      staleTime: props.refreshTime ?? REFETCH_TIME,
+    })),
+    combine: results => {
+      return {
+        data: results.flatMap(result => result.data ?? []),
+        pending: results.some(result => result.isPending),
+        loading: results.some(result => result.isLoading),
+        hasError: results.some(result => result.isError),
+      };
```

Comment:
this type of customisation isn't ideal, could we try to remain consistent on names ie. if it is `isError` then let's keep this instead of hasError ...